docs/index.adoc

```diff
@@ -218,139 +218,180 @@ libStatic.buildGetter('my/folder', {
+
+{zwsp} +
 
+All the previous examples use lib-static in link:https://developer.enonic.com/docs/xp/stable/runtime/engines/http-service[XP services], because services act exactly like this. The premises are fulfilled out of the box here, so the path resolution works without further setup.
+
+However, lib-static can be used in other contexts than in services, where these premises may not be true and you may need to roll your own path resolution:
+
+{zwsp} +
 
 [[example-getcleanpath]]
-==== a) getCleanPath
-However, there can be cases where you need to customize the relative-asset-path resolution - for example, using an link:https://developer.enonic.com/docs/xp/stable/cms/mappings[XP controller mapping] for setting up an endpoint that uses lib-static.
+==== a) Override: getCleanPath
 
-Send an link:#example-options[option] function `getCleanPath` to `.buildGetter`. `getCleanPath` takes the `request` argument and returns a relative asset path. The rest is up to you:
+You can *override the default file path resolution* by implementing a `request => string` function, and add that as a `getCleanPath` option in `.buildGetter`.
+
+👉 link:#options[Options API reference]
+
+For example, a simplified version of the default could be implemented like this:
 
 [source,javascript,options="nowrap"]
 ----
 exports.get = libStatic.buildGetter({
     root: 'my/folder',
-    getCleanPath: function(request) {
-        // In a perfect imaginary example world, all requests handled here have a
-        // request.path (URL after the domain) that start with '/i/am/a/prefix`, so just
-        // remove it, hardcoded, to get the correct relative path.
-        // You should probably make more of an effort though:
-        return request.path.substring('/i/am/a/prefix'.length);
+    getCleanPath: request => {
+        const prefix = request.contextPath;
+        return request.rawPath.substring(prefix.length);
     }
-
-    // Request path: **/i/am/a/prefix/subdir/myFile.txt
-    // --> Relative resource path: subdir/myFile.txt
-    // --> lib-static looks up my/folder/subdir/myFile.txt
 });
 ----
 
-...or...
+{zwsp} +
 
-[source,javascript,options="nowrap"]
+[[example-gotchas]]
+==== b) Gotchas
+
+When writing a good `getCleanPath` function, here are some *rules of thumb*:
+
+1. `request => string` function, where the string is the final resolved relative path under `<root>`
+2. In order for index fallbacks to work properly:
+  - URIs with a trailing slash should also return the trailing slash in the string,
+  - And vice versa: URIs without a trailing slash should not return one,
+  - URIs to the endpoint itself should return an empty string (unless there's a trailing slash, in which case only the slash should be returned),
+  - URIs to other content below the endpoint should return a path beginning with a slash.
+3. Use `request.rawPath` as the basis. Don't use `request.path`.
```

Comment:
Albeit it is true true, we may change this in XP quite soon https://github.com/enonic/xp/issues/8848

Reply:
But that's only the trailing slash thing isn't it? The vhosting and special-chars will stay different between `path` and `rawPath`, right?

Reply:
true